lobby/src/test/java/org/triplea/lobby/server/login/LobbyLoginValidatorIntegrationTest.java

```diff
@@ -134,13 +147,24 @@ void testLogin() {
                 new HashedPassword(BCrypt.hashpw(hashPasswordWithSalt(password), BCrypt.gensalt())))
             .apply(
                 challenge -> {
-                  response.putAll(RsaAuthenticator.newResponse(challenge, password));
+                  response.put(
+                      LobbyLoginResponseKeys.RSA_ENCRYPTED_PASSWORD,
+                      RsaAuthenticator.encrpytPassword(
+                          challenge.get(LobbyLoginChallengeKeys.RSA_PUBLIC_KEY), password));
                   return response;
                 }));
     // with a bad password
     assertError(
         generateChallenge(user, null)
-            .apply(challenge -> new HashMap<>(RsaAuthenticator.newResponse(challenge, "wrong"))),
+            .apply(
+                challenge -> {
+                  final Map<String, String> map = new HashMap<>();
+                  map.put(
+                      LobbyLoginResponseKeys.RSA_ENCRYPTED_PASSWORD,
+                      RsaAuthenticator.encrpytPassword(
+                          challenge.get(LobbyLoginChallengeKeys.RSA_PUBLIC_KEY), "wrong"));
+                  return map;
+                }),
```

Comment:
Collections.singletonMap ?

Reply:
The test mutates the collection that is returned, line 59:
```
    return responseGetter -> {
      final Map<String, String> response = responseGetter.apply(challenge);
      response.putIfAbsent(LobbyLoginResponseKeys.EMAIL, EMAIL);
      response.putIfAbsent(
          LobbyLoginResponseKeys.LOBBY_VERSION, LobbyConstants.LOBBY_VERSION.toString());
      return loginValidator.verifyConnection(challenge, response, name, mac, address);
    };
```
This test could use some work to be more explicit, probably the map should be a constant instead of something derived by a function. Less DRY and more copy/paste would be good for this test IMO.

Reply:
Well, you could wrap it in a HashMap to get a one liner, but whatever not that important anyways.
This is one of the cases where C# has superior language features like initializers that would allow such things to be done per implementation